rust/tools/authorization-logic/src/ast.rs

```diff
@@ -28,6 +28,7 @@ pub struct AstPrincipal {
 /// implemented manually.
 #[derive(Debug, Hash, Clone, Serialize, Deserialize)]
 pub struct AstPredicate {
+    pub sign: bool, // false if negatedm, true otherwise
```

Comment:
This seems like a source of confusion. There are two ways to naively read this field when you're far away from the field's definition. One is "Is this value signed (ie, negative when true)" the other is "What is the polarity of this sign (ie, positive when true)". To prevent bugs, it might be worth creating an `enum` with two values: `Negated` and `NotNegated` to make the intent clear at the various usage sites.

Also, Nit: typo in comment.

Reply:
ok fixed.